fs/nfsd/vfs.c

```diff
@@ -1303,7 +1317,7 @@ nfsd_create_locked(struct svc_rqst *rqstp, struct svc_fh *fhp,
 		iap->ia_mode = 0;
 	iap->ia_mode = (iap->ia_mode & S_IALLUGO) | type;
 
-	if (!IS_POSIXACL(dirp))
+	if (!IS_POSIXACL(dirp) && !IS_NFSV4ACL(dirp))
 		iap->ia_mode &= ~current_umask();
 
```

Comment:
I Linux 5.15, there was `do_nfsd_create` for NFSv3 and NFSv4 version of `nfsd_create`. `do_nfsd_create` was updated just like here and ` && !IS_NFSV4ACL(dirp)` was added there as well for v5.15.

In Linux 6.1, there is `nfsd3_create_file` for NFSv3 and `nfsd4_create_file` for NFSv4. It looks like both these functions replace `do_nfsd_create`. Both these functions contain:
```
	if (!IS_POSIXACL(inode))
		iap->ia_mode &= ~current_umask();
```
Should ` && !IS_NFSV4ACL(inode)` be added to `nfsd3_create_file` and `nfsd4_create_file`?

Reply:
Yes. Good catch!